backend/src/main/java/com/bakdata/conquery/models/events/EmptyStringType.java

```diff
@@ -0,0 +1,92 @@
+package com.bakdata.conquery.models.events;
+
+import java.util.Collections;
+import java.util.Iterator;
+
+import com.bakdata.conquery.io.cps.CPSType;
+import com.bakdata.conquery.models.dictionary.Dictionary;
+import com.bakdata.conquery.models.events.stores.ColumnStore;
+import com.bakdata.conquery.models.events.stores.specific.string.StringType;
+import com.bakdata.conquery.models.identifiable.ids.specific.DictionaryId;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * An empty generic StringStore to avoid any allocations. It still has a length, but {@linkplain #has(int)}} is always false.
+ */
+@CPSType(base = ColumnStore.class, id = "EMPTY_STRING")
+public class EmptyStringType extends StringType {
```

Comment:
Jetzt ist der EmptyStringType ein weiterer Spezialfall. Was spricht gegen eine Fallunterscheidung in https://github.com/bakdata/conquery/blob/db397d638d5bbc5ac176c1ee086ef7d2dc533adc/backend/src/main/java/com/bakdata/conquery/models/concepts/filters/specific/AbstractSelectFilter.java#L85 ?

Reply:
Ich hatte mal in die QueryEngine geschaut, und da gibt es viele Aggregatoren die einfach nen cast machen, deswegen erschien mir das sinnvoller. Aber auch doof. Man könnte auch die StrinStore spezialisierung wieder hoch ziehen wie bei der alten Implementierung wo es dann halt in vielen Fällen ungenutzt war.

Reply:
Also ich hab gerade so 1-2h damit rumgespielt den ColumnStore aufzuspalten in eine ober klasse, und dann spezifischen Unterklassen, und das wird noch etwas aufwändiger aber deutlich schöner, weil es dann komplett Typesafe ist und keine komischen Generics mehr rumschwirren.